### ISSUE TITLE
Split up GDScript basics into many smaller sections.

### ISSUE THREAD

**Issue by golddotasksquestions** [godotengine/godot-docs#2735]
https://docs.godotengine.org/en/3.1/getting_started/scripting/gdscript/gdscript_basics.html#doc-gdscript
This page is simply overwhelming. I never read it in full, only look up bits and pieces here or there when I need.
Splitting off exports (https://github.com/godotengine/godot-docs/pull/2718) is into one (or more) pages(s) is definitely a good idea, but can we have this for more of it's topics?

While it's called "Basics" of GDScript it actually seems like quite a comprehensive collection of it's features that go way beyond what I would consider basic. It's incredible valuable, overall very well explained information, but if was structured into smaller junks with pages (and more detailed list of the topics to left for each page), I would not get intimidated that easy, plus quickly get an overview about what information I can find here.

**Comment by asheraryam** [comment]
I don't think we _need_ to split up the page, but I think it's a general problem that there's no Table of Contents for large pages like this.

**Comment by golddotasksquestions** [comment]
If there was a only table of content, it would stretch over multiple screens and just make this page larger.

Another problem here is that is you online search for something, like how to instance a scene for example the search result will just be this page and you will have to perform another search within that page to get what you are looking for. Not the best solution either.

**Comment by Meriipu** [comment]
I really liked having it all on one page when I first read it, and I still enjoy it those times where I return to it to look something up.
it is easy for me to get lost when there are tons of nested pages or many pages in a list each with barely any content. 
It leads to a lot of backwards/forwards or a lot of tabs.
It also makes it very hard to ctrl+f to search.

**Comment by Calinou** [comment]
@Meriipu There's a middle ground to find for sure, but I think having complex GDScript concepts on separate pages makes more sense still.

**Comment by Haledire** [comment]
Didn't notice this issue before - so I'll bring this up here
yield is going to be removed in favor of await
yield is a function - await is a keyword
Is the method for coroutines going to be removed from documentation available offline and available strictly from the online docs? it's currently visible in @gdscript because it's a method and not a keyword.

I do feel there's some merit to handling something for offline visibility of the documentation handled in the basics page.

**Comment by Calinou** [comment]
@Haledire Keep in mind you can download offline HTML builds of the documentation from [this page](hugo.pro/projects/godot-builds/).

**Comment by skyace65** [comment]
The GDScript basics page got renamed to GDScript reference at some point. Given that it's a reference page now and not an introduction does it still need to be split up into multiple pages?

**Comment by golddotasksquestions** [comment]
@skyace65 Hm, Idk. I still think this page is unreasonable long. Too long too take it all in at once. But calling it "Reference" is definitely more accurate.
